app/src/main/java/com/example/cinemapp/ui/main/profile/ProfileViewModel.kt

```diff
@@ -94,32 +106,51 @@ class ProfileViewModel(
         getNextPage(MovieRepository.MovieListType.RATED)
     }
 
-    fun getNextPage(movieListType: MovieRepository.MovieListType = state.value.movieListType, resetPaging: Boolean = false) {
-        if (!isPaging) {
-            setPagingRunning(true)
-            viewModelScope.launch {
-                val sessionId = userPrefs.getSessionId().firstOrNull()
-                sessionId?.let {
+    fun getNextPage(
+        movieListType: MovieRepository.MovieListType = state.value.movieListType,
+        resetPaging: Boolean = false
+    ) {
+        viewModelScope.launch {
+            val sessionId = userPrefs.getSessionId().firstOrNull()
+            sessionId?.let {
+                if (!isPaging) {
+                    setPagingRunning(true)
                     _state.update {
-                        val newPage = if (resetPaging) 1 else if (movieListType == it.movieListType) it.pagesLoaded + 1 else 1
-                        val listInfo = movieRepository.getMovieList(
-                            movieListType,
-                            newPage,
-                            sessionId = sessionId
-                        )?.let { listInfo -> movieListMapper.mapToCardListInfo(listInfo, 400) }
-                            ?: MovieListInfo()
-                        it.copy(
-                            pagesLoaded = newPage,
-                            movieListType = movieListType,
-                            movies = listInfo.results,
-                            totalPages = listInfo.totalPages
-                        )
+                        getNextPageData(movieListType, resetPaging, sessionId).first()
                     }
-                } ?: setPagingRunning(false)
+                }
             }
         }
     }
 
+    private fun getNextPageData(
+        movieListType: MovieRepository.MovieListType = state.value.movieListType,
+        resetPaging: Boolean = false,
+        sessionId: String
+    ): Flow<State> {
+        return flow {
```

Comment:
There is no need to mimic a flow here, since you are returning only a single value.

Reply:
Actually it's returning multiple values that changed like list type, number of page and also the list itself.